libraries/psibase/common/src/Rpc.cpp

```diff
@@ -82,22 +82,31 @@ namespace psibase
    }
 
    std::optional<std::string_view> HttpRequest::getCookie(std::string_view name) const
+   {
+      if (auto cookieHeader = getHeader("cookie"))
+      {
+         for (auto kvrange : *cookieHeader | std::views::split(';'))
+         {
+            std::string_view kv  = split2sv(kvrange);
+            auto             pos = kv.find('=');
+            check(pos != std::string_view::npos, "Invalid cookie");
+            auto leading = kv.find_first_not_of(" \t");
+            auto key     = kv.substr(leading, pos - leading);
+            auto value   = kv.substr(pos + 1);
+            if (key == name)
+               return value;
+         }
+      }
+      return {};
+   }
+
+   std::optional<std::string_view> HttpRequest::getHeader(std::string_view name) const
    {
       for (const auto& header : headers)
       {
-         if (std::ranges::equal(header.name, std::string_view{"cookie"}, {}, ::tolower))
+         if (std::ranges::equal(header.name, name, {}, ::tolower))
```

Comment:
`getHeader` should not assume that the name passed in is lower case. You can use `header.matches`.